polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -2256,3 +2506,40 @@ fn descriptor_version_sanity_check(
 		descriptor_version => Err(SecondingError::InvalidReceiptVersion(descriptor_version)),
 	}
 }
+
+// Checks that there are enough free connection slots for AssetHub invulnerable collators.
+fn can_accept_new_collator_connection(
+	new_peer_id: &PeerId,
+	currently_connected: &HashMap<PeerId, PeerData>,
+	ah_invulnerables: &HashSet<PeerId>,
+) -> bool {
+	if ah_invulnerables.contains(new_peer_id) {
+		// If the new peer is AH invulnerable, we always accept it.
```

Comment:
also if the invulnerables list is empty, no?

I'm a bit confused about why we initially express the invulnerables list as `Option<HashSet>` and later transform it into a `HashSet`.

Reply:
For the `Option` part:
- `Some` means that a list was provided with a command line param. We use whatever was provided in the file.
- `None` means no option was probvided. We use the hardcoded list.

On why convert it to `HashSet` - once we know if we should use the hardcoded list or not the `Option` is no longer useful. This was the safety net I was talking about. It's complicated a bit because I wanted everything in one option and it was not intended for regular use.

But as we agreed - this all will go away.